src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -95,16 +99,22 @@
     private static final MultiThreadedHttpConnectionManager connectionManager = new MultiThreadedHttpConnectionManager();
     private final String owner;
     private final String repositoryName;
-    private final UsernamePasswordCredentials credentials;
+    private final Credentials credentials;
```

Comment:
I would much rather switch to using the AuthenticationTokens API to handle this rather than adding more complex fields and logic.

Ideally you'd have a `BitbucketAuthenticator` abstract class and then have `AuthenticationTokens` convert from credentials to this `BitbucketAuthenticator`... then the `BitbucketAuthenticator` would be handed the `HttpClient` to inject its authentication

Reply:
ok, let me look in to it.